lib/mof/helper.rb

```diff
@@ -67,32 +67,48 @@ def open name, origin = nil
 	raise "Cannot open #{name}"
       end
     end
-    @fstack << [ @file, @name, @lineno, @iconv, $/, @result ] if @file  
+    @fstack << [ @file, @name, @lineno, @iconv, $/, @result ] if @file
     @file = file
     @name = name
     @lineno = 1
+    @iconv = nil
+    $/ = "\n"
+
     @result = MOF::Result.new
     # read the byte order mark to check for utf-16 windows files
     bom = @file.read(2)
-    if bom == "\376\377"
+
+    # check for 2-byte BOMs
+    if bom.bytes == [0xFE, 0xFF]
```

Comment:
In Ruby 1.9.3 `bytes` is an enumerator type whereas in 2.x `bytes` returns an array

Suggest the following changes;
Line 79 = `bom = (@file.read(2)).each_byte.to_a`
All instances of `bom.bytes` to `bom`
Line 82 = `bom += (@file.read(1)).each_byte.to_a`

http://ruby-doc.org/core-2.2.3/String.html#method-i-bytes
http://ruby-doc.org/core-1.9.3/String.html#method-i-bytes